src/main/java/duke/data/Storage.java

```diff
@@ -0,0 +1,147 @@
+package duke.data;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Scanner;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+import duke.ui.Ui;
+
+/**
+ * Represents storage that can store data into a file and read data from a file.
+ */
+public class Storage {
+    /** Path of the current folder as a string */
+    public static final String DIRECTORY_PATH = System.getProperty("user.dir");
+    /** Path of file containing data saved */
+    private static File data;
+    /** A temporary list of tasks */
+    private TaskList tasks;
+
+    /**
+     * Constructor of the `Storage` class.
+     *
+     * @param filePath Path to the file to be loaded.
+     * @param tasks A temporary list of tasks.
+     */
+    public Storage(Path filePath, TaskList tasks) {
+        Storage.data = filePath.toFile();
+        this.tasks = tasks;
+        this.readFile();
+    }
+
+    /**
+     * Reads data in the file. If the directory or file doesn't exist, create a new file in the correct
+     * location.
+     */
+    private void readFile() {
+        try {
+            Scanner fileScanner = new Scanner(Storage.data);
+            while (fileScanner.hasNextLine()) {
+                String task = fileScanner.nextLine();
+                this.readData(task);
+            }
+            fileScanner.close();
+        } catch (FileNotFoundException fileNotFoundException) { // If file doesn't exist, create it.
```

Comment:
Should the comment be in a separate line from the code?
This is not explicitly stated in the Java coding standard but can help with consistency with the rest of the code.
Suggestion :
```java
catch (FileNotFoundException fileNotFoundException) { 
    // If file doesn't exist, create it.
```